Workbooks/Traffic Analytics/Unified/ApplicationPorts.workbook

```diff
@@ -0,0 +1,955 @@
ption = tostring(hostSplit[0]), subnet = tostring(hostSplit[1])\r\n// fork part 1 Flows:\r\n    | extend \r\n    AllowedTraffic = tolong(AllowedInbound) + tolong(AllowedOutbound),\r\n    BlockedTraffic = tolong(BlockedInbound) + tolong(BlockedOutbound)\r\n    // fork part 2 Bytes:\r\n    | extend \r\n    AllowedTrafficBytes = tolong(AllowedInboundBytesAtDest) + tolong(AllowedInboundBytesAtSrc) + tolong(AllowedOutboundBytesAtSrc) + tolong(AllowedOutboundBytesAtDest),\r\n    BlockedTrafficBytes = tolong(BlockedInboundBytesAtDest) + tolong(BlockedInboundBytesAtSrc) + tolong(BlockedOutboundBytesAtSrc) + tolong(BlockedOutboundBytesAtDest) \r\n    // fork part 3 Packets:\r\n    | extend \r\n    AllowedTrafficPackets = tolong(AllowedInboundPacketsAtDest) + tolong(AllowedInboundPacketsAtSrc) + tolong(AllowedOutboundPacketsAtSrc) + tolong(AllowedOutboundPacketsAtDest),\r\n    BlockedTrafficPackets = tolong(BlockedInboundPacketsAtDest) + tolong(BlockedInboundPacketsAtSrc) + tolong(BlockedOutboundPacketsAtSrc) + tolong(BlockedOutboundPacketsAtDest)\r\n    | summarize \r\n    // summarize for flows\r\n    AllowedTraffic = sum(AllowedTraffic), \r\n    BlockedTraffic = sum(BlockedTraffic),\r\n    MaliciousFlow = sum(iff(isMaliciousFlow, AllowedTraffic + BlockedTraffic, 0)),\r\n    TotalFlow = sum(FlowCount),\r\n    // summarize for bytes\r\n    AllowedBytes = sum(AllowedTrafficBytes), \r\n    BlockedBytes = sum(BlockedTrafficBytes),\r\n    MaliciousBytes = sum(iff(isMaliciousFlow, AllowedTrafficBytes + BlockedTrafficBytes, 0)),\r\n    TotalBytes = sum(BytesCount),\r\n    // summarize for packets\r\n    AllowedPackets = sum(AllowedTrafficPackets),\r\n    BlockedPackets = sum(BlockedTrafficPackets),\r\n    MaliciousPackets = sum(iff(isMaliciousFlow, AllowedTrafficPackets + BlockedTrafficPackets, 0)),\r\n    TotalPackets = sum(PacketsCount) \r\n    by subscription,subnet//IP, VM, Subnet, Subscription\r\n| summarize totalTraffic = iff('{Units:value}' =~ 'bytes', sum(TotalBytes), iff('{Units:value}' =~ 'packets', sum(TotalPackets), sum(TotalFlow))),\r\nallowedTraffic = iff('{Units:value}' =~ 'bytes', sum(AllowedBytes), iff('{Units:value}' =~ 'packets', sum(AllowedPackets), sum(AllowedTraffic))),\r\nmaliciousTraffic = iff('{Units:value}' =~ 'bytes', sum(MaliciousBytes), iff('{Units:value}' =~ 'packets', sum(MaliciousPackets), sum(MaliciousFlow))),\r\nblockedTraffic = iff('{Units:value}' =~ 'bytes', sum(BlockedBytes), iff('{Units:value}' =~ 'packets', sum(BlockedPackets), sum(BlockedTraffic)));\r\n union (\r\n        subnetTraffic\r\n        | project ColumnName = 'TotalTraffic', Value = totalTraffic\r\n    ),\r\n    (\r\n        subnetTraffic\r\n        | project ColumnName = 'AllowedTraffic', Value = allowedTraffic\r\n    ),\r\n    (\r\n        subnetTraffic\r\n        | project ColumnName = 'Malicious Traffic', Value = maliciousTraffic\r\n    ),\r\n    (\r\n        subnetTraffic\r\n        | project ColumnName = 'Blocked Traffic', Value = blockedTraffic\r\n    )",
```

Comment:
why is this time hard-coded in the query?
(datetime('2023-07-08T21:38:19.028Z') .. datetime('2023-07-09T21:38:19.028Z'))

Reply:
Removed.